test/ctia/bundle/core_test.clj

```diff
@@ -56,38 +56,56 @@
-                           (constantly
-                            (map (fn [old-id]
-                                   {:entity {:id old-id}})
-                                 existing-ids)))
+                         :external_ids ["swe-alarm-indicator-1"]}
+          find-by-ext-ids (fn [existing-ids]
+                            (constantly
+                             (map (fn [old-id]
+                                    {:entity {:id old-id}})
+                                  existing-ids)))
           test-fn (fn [{:keys [msg expected existing-ids log?]}]
                     (with-log
                       (testing msg
                         (is (= expected
                                (sut/with-existing-entity
                                  new-indicator
-                                 (find-by-ext-id existing-ids))))
+                                 (find-by-ext-ids existing-ids))))
                         (is (= log?
                                (logged? 'ctia.bundle.core
                                         :warn
-                                        #"More than one entity is linked to the external id swe-alarm-indicator-1"))))))]
+                                        #"More than one entity is linked to the external ids"))))))]
       (test-fn {:msg "no existing external id"
                 :expected {:id indicator-id-3
-                           :external_id "swe-alarm-indicator-1"}
+                           :external_ids ["swe-alarm-indicator-1"]}
                 :existing-ids []
                 :log? false})
       (test-fn {:msg "1 existing external id"
                 :expected (with-long-id {:result "exists"
-                                         :external_id "swe-alarm-indicator-1"
+                                         :external_ids ["swe-alarm-indicator-1"]
                                          :id indicator-id-1})
                 :existing-ids [indicator-id-1]
                 :log? false})
       (test-fn {:msg "more than 1 existing external id"
                 :expected (with-long-id {:result "exists"
-                                         :external_id "swe-alarm-indicator-1"
+                                         :external_ids ["swe-alarm-indicator-1"]
                                          :id indicator-id-2})
                 :existing-ids [indicator-id-2
                                indicator-id-1]
                 :log? true}))))
+
+(deftest filter-external-ids-test
+  (let [external-ids ["ctia-indicator-1" "cisco-indicator-1" "indicator-1"]
+        test-fn (fn [prefixes expected log?])]
+    (are [expected prefixes log?]
+        (testing prefixes
+          (with-log
+            (is (= expected
+                   (sut/filter-external-ids external-ids prefixes)))
+            (is (= log?
+                   (logged? 'ctia.bundle.core
+                            :warn
+                            #"More than 1 valid external ID has been found")))))
```

Comment:
remember to place a `true` at tail position

I've updated my Yasnipet snippet to include comments explaining each used technique:

https://raw.githubusercontent.com/zenmacs/.emacs.d/b8774255f53212eca7d128d614c5dd159679e55a/snippets/clojure-mode/are